test/e2e/storage/testsuites/multivolume.go

```diff
@@ -0,0 +1,312 @@
+		}()
+		Expect(err).NotTo(HaveOccurred())
+		pod1, err = l.cs.CoreV1().Pods(pod1.Namespace).Get(pod1.Name, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred(), "get pod1")
+		actualNodeName := pod1.Spec.NodeName
+
+		// Set affinity depending on requiresSameNode
+		if requiresSameNode {
+			framework.SetAffinity(&node, actualNodeName)
+		} else {
+			framework.SetAntiAffinity(&node, actualNodeName)
+		}
+
+		By(fmt.Sprintf("Creating 2nd pod with a volume on %+v", node))
+		pod2, err := framework.CreateSecPodWithNodeSelection(l.cs, l.ns.Name,
+			[]*v1.PersistentVolumeClaim{l.resource2.pvc},
+			false, "", false, false, framework.SELinuxLabel,
+			nil, node, framework.PodStartTimeout)
+		defer func() {
+			framework.ExpectNoError(framework.DeletePodWithWait(f, l.cs, pod2))
+		}()
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Checking if the volume in 1st pod exists as expected volume mode")
+		utils.CheckVolumeModeOfPath(pod1, l.resource1.pattern.VolMode, "/mnt/volume1")
+
+		By("Checking if read/write to the volume in 1st pod works properly")
+		utils.CheckReadWriteToPath(pod1, l.resource1.pattern.VolMode, "/mnt/volume1")
+
+		By("Checking if the volume in 2nd pod exists as expected volume mode")
+		utils.CheckVolumeModeOfPath(pod2, l.resource2.pattern.VolMode, "/mnt/volume1")
+
+		By("Checking if read/write to the volume in 2nd pod works properly")
+		utils.CheckReadWriteToPath(pod2, l.resource2.pattern.VolMode, "/mnt/volume1")
+
+		// Delete pod2
+		framework.ExpectNoError(framework.DeletePodWithWait(f, l.cs, pod2))
+
+		// Test read/write work well in pod1 after pod2 is deleted
+		By("Rechecking if the volume in 1st pod exists as expected volume mode")
+		utils.CheckVolumeModeOfPath(pod1, l.resource1.pattern.VolMode, "/mnt/volume1")
+
+		By("Rechecking if read/write to the volume in 1st pod works properly")
+		utils.CheckReadWriteToPath(pod1, l.resource1.pattern.VolMode, "/mnt/volume1")
+	}
+
+	// This tests below configuration:
+	//          [pod1]
+	//          /    \      <- same volume mode
+	//   [volume1]  [volume2]
+	It("should attach two volumes with the same volume mode to the same pod", func() {
+		// Currently, multiple volumes are not generally available for pre-provisoined volume,
+		// because containerized storage servers, such as iSCSI and rbd, are just returning
+		// a static volume inside container, not actually creating a new volume per request.
+		if pattern.VolType == testpatterns.PreprovisionedPV {
+			framework.Skipf("This test doesn't work with pre-provisioned volume -- skipping")
+		}
+
+		init()
+		defer cleanup()
+
+		l.resource2.pattern.VolMode = l.resource1.pattern.VolMode
+		testAttachTwoVolumesToOnePod(l)
+	})
+
+	// This tests below configuration:
+	//          [pod1]
+	//          /    \      <- different volume mode (only <block, filesystem> pattern is tested)
+	//   [volume1]  [volume2]
+	It("should attach two volumes with different volume mode to the same pod", func() {
```

Comment:
in the test case names maybe replace "attach" with "access"? to not confuse with the actual attach operation